src/DebugEngineHost/RegistryMonitor.cs

```diff
@@ -80,10 +77,7 @@ public void Start()
 
         public void Stop()
         {
-            if (m_stoppedEvent != null)
-            {
-                m_stoppedEvent.Set();
-            }
+            _stoppedEvent?.Set();
```

Comment:
>_stoppedEvent?.Set(); [](http://example.com/codeflow?start=12&length=21)

Is it possible to call `Stop` more than once? If so, this could throw `ObjectDisposedException`.

Reply:
I think we should be able to call stop multiple times safely. 

However, I think we should only allow Dispose to be called once?

Reply:
If you want to allow calling `Stop` more than once, you need something like an `isStopped` field and take a lock or do an interlocked operation in order to make sure that `_stoppedEvent.Set()` is only called once. Calling it a second time may throw since the other thread disposes it.